egs/wsj/s5/steps/make_index.sh

```diff
@@ -72,14 +71,26 @@ if [ ! -z $silence_word ]; then
   silence_opt="--silence-label=$silence_int"
 fi
 
-$cmd JOB=1:$nj $kwsdir/log/index.JOB.log \
-  lattice-add-penalty --word-ins-penalty=$word_ins_penalty "ark:gzip -cdf $decodedir/lat.JOB.gz|" ark:- \| \
-    lattice-align-words $silence_opt --max-expand=$max_expand $word_boundary $model  ark:- ark:- \| \
-    lattice-scale --acoustic-scale=$acwt --lm-scale=$lmwt ark:- ark:- \| \
-    lattice-to-kws-index --max-states-scale=$max_states_scale --allow-partial=true \
-    --max-silence-frames=$max_silence_frames --strict=$strict ark:$utter_id ark:- ark:- \| \
-    kws-index-union --skip-optimization=$skip_optimization --strict=$strict --max-states=$max_states \
-    ark:- "ark:|gzip -c > $kwsdir/index.JOB.gz" || exit 1
-    
+word_boundary=$langdir/phones/word_boundary.int
+if [ -f $word_boundary ] ; then
+  $cmd JOB=1:$nj $kwsdir/log/index.JOB.log \
+    lattice-add-penalty --word-ins-penalty=$word_ins_penalty "ark:gzip -cdf $decodedir/lat.JOB.gz|" ark:- \| \
+      lattice-align-words $silence_opt --max-expand=$max_expand $word_boundary $model  ark:- ark:- \| \
+      lattice-scale --acoustic-scale=$acwt --lm-scale=$lmwt ark:- ark:- \| \
+      lattice-to-kws-index --max-states-scale=$max_states_scale --allow-partial=true \
+      --max-silence-frames=$max_silence_frames --strict=$strict ark:$utter_id ark:- ark:- \| \
+      kws-index-union --skip-optimization=$skip_optimization --strict=$strict --max-states=$max_states \
+      ark:- "ark:|gzip -c > $kwsdir/index.JOB.gz" || exit 1
+else
+  align_lexicon=$langdir/phones/align_lexicon.int 
```

Comment:
At this point you should check that align_lexicon exists.    It may not if people used an out-of-date prepare_lang.sh.